addon/-private/system/model/record-data-map.ts

```diff
@@ -0,0 +1,54 @@
+import { DEBUG } from '@glimmer/env';
+
+export const RECORD_DATA_KEY = `${Date.now()}-record-data`;
+
+// TODO is there a way to import classes without causing cyclical imports for rollup?
+//   maybe strip "typing" imports prior to rollup?
+type InternalModel = object;
+type CreateOptions = object;
+type Record = object;
+type RecordData = {
+  hasAttr(key);
+  getAttr(key);
+  setDirtyAttribute(key, value);
+  isAttrDirty(key);
+};
+
+type Mappable = InternalModel | CreateOptions | Record;
+
+const RecordDataMap = new WeakMap<Mappable, RecordData>();
+
+export function getRecordDataFor(instance: Mappable): RecordData {
```

Comment:
Can you add some comments to the file please, what the purpose is and how to use it